src/main/java/com/o19s/es/ltr/action/ClearCachesAction.java

```diff
@@ -0,0 +1,177 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.o19s.es.ltr.action;
+
+import org.elasticsearch.action.Action;
+import org.elasticsearch.action.ActionRequestBuilder;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.FailedNodeException;
+import org.elasticsearch.action.support.nodes.BaseNodeResponse;
+import org.elasticsearch.action.support.nodes.BaseNodesRequest;
+import org.elasticsearch.action.support.nodes.BaseNodesResponse;
+import org.elasticsearch.client.ElasticsearchClient;
+import org.elasticsearch.cluster.ClusterName;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class ClearCachesAction extends Action<ClearCachesAction.ClearCachesNodesRequest,
+        ClearCachesAction.ClearCachesNodesResponse, ClearCachesAction.RequestBuilder> {
+    public static final String NAME = "ltr:caches";
+    public static final ClearCachesAction INSTANCE = new ClearCachesAction();
+
+    private ClearCachesAction() {
+        super(NAME);
+    }
+
+    @Override
+    public RequestBuilder newRequestBuilder(ElasticsearchClient client) {
+        return new RequestBuilder(client);
+    }
+
+    @Override
+    public ClearCachesNodesResponse newResponse() {
+        return new ClearCachesNodesResponse();
+    }
+
+    public static class RequestBuilder extends ActionRequestBuilder<ClearCachesNodesRequest, ClearCachesNodesResponse,
+            ClearCachesAction.RequestBuilder> {
+        RequestBuilder(ElasticsearchClient client) {
+            super(client, ClearCachesAction.INSTANCE, new ClearCachesNodesRequest());
+        }
+    }
+
+    public static class ClearCachesNodesRequest extends BaseNodesRequest<ClearCachesNodesRequest> {
+        private String store;
+        private Operation operation;
+        private String name;
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException arve = null;
+            if (store == null) {
+                arve = addValidationError("store cannot be null", null);
+            }
+
+            if (operation == null) {
+                arve = addValidationError("no operation provided", arve);
+            }
+
+            if (operation != null && operation != Operation.ClearStore && name == null) {
```

Comment:
in fact it can, we do not fail on the previous if and here we continue to collect validation errors so we must check against null not to add a second error.